runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessActionVariables.ts

```diff
@@ -1,27 +1,85 @@
 import { useMemo } from 'react';
 
+import { addScope, getAncestorScope, getAncestorScopesActions } from '#src-app/utils/variableScopes';
 import {
-    BPMNElement,
     ExtensionElement,
     ModdleElement,
 } from '#src-app/views/process/ProcessBuildView/Modeler/helpers/elementParameters';
 
 import { useModelerContext } from './useModelerContext';
-
 import {
     ActionVariableObject,
     ActionVariables,
+    Scope,
+    ScopedModdleElement,
 } from './useProcessVariables.types';
 
-const useProcessActionVariables = () => {
+
+const useProcessActionVariables = (selectedElementParentId?: string) => {
     const context = useModelerContext();
     const canvas = context?.modeler?.get('canvas');
     const rootElement = canvas?.getRootElement();
-    const allActionsWithVariables: ModdleElement[] =
-        rootElement?.businessObject?.flowElements.filter((item: BPMNElement) =>
+    const rootElementScopeId = rootElement?.businessObject?.id;
+    const ProcessScopes: Scope = useMemo<Scope>(() => ({ id: rootElementScopeId }), [rootElementScopeId]);
```

Comment:
adjust variable naming convention + useMemo here is pointless

Reply:
done
allVarsActions forces useMemo